File: src/main/kotlin/school/lec07/Lec07Main.java
package school.lec07;

import org.jetbrains.annotations.NotNull;

import java.io.IOException;

public class Lec07Main {
    public static void main(String[] args) throws IOException {
        JavaFilePrinter printer = new JavaFilePrinter();
        printer.readFile();
    }

    /*private static int parseIntOrThrow(@NotNull String str){
        try{
            return Integer.parseInt(str);

        } catch (NumberFormatException e){
            throw new IllegalArgumentException(String.format("주어진 %s는 숫자가 아닙니다",str));
        }
    }*/




}

File: src/main/kotlin/school/lec07/Lec07Main.kt
package school.lec07

import java.lang.NumberFormatException

fun parseIntOrThrow(str : String):Int{
    try {
        return str.toInt()
    }catch (e:NumberFormatException){
        throw IllegalArgumentException("주어진 ${str}는 숫자가 아닙니다")
    }
}